core/src/main/java/org/elasticsearch/discovery/zen/ZenDiscovery.java

```diff
@@ -922,7 +927,7 @@ private DiscoveryNode findMaster() {
                 return winner.getNode();
             } else {
                 // if we don't have enough master nodes, we bail, because there are not enough master to elect from
-                logger.trace("not enough master nodes [{}]", masterCandidates);
+                logger.warn("not enough master nodes [{}], but needed [{}]", masterCandidates, electMaster.minimumMasterNodes());
```

Comment:
since we promote this to first class message, I think we should make it more user friendly. how about "not enough master nodes discovered during pinging (found [{}], needed [{}]), pinging again" ?